packages/database/src/core/SyncPoint.ts

```diff
@@ -41,239 +39,254 @@ let __referenceConstructor: ReferenceConstructor;
  *  - Proxying user / server operations to the views as appropriate (i.e. applyServerOverwrite,
  *    applyUserOverwrite, etc.)
  */
-export class SyncPoint {
-  static set __referenceConstructor(val: ReferenceConstructor) {
-    assert(
-      !__referenceConstructor,
-      '__referenceConstructor has already been defined'
-    );
-    __referenceConstructor = val;
-  }
 
-  static get __referenceConstructor() {
-    assert(__referenceConstructor, 'Reference.ts has not been loaded');
-    return __referenceConstructor;
-  }
+let referenceConstructor: ReferenceConstructor;
 
+export function syncPointSetReferenceConstructor(
+  val: ReferenceConstructor
+): void {
+  assert(
+    !referenceConstructor,
+    '__referenceConstructor has already been defined'
+  );
+  referenceConstructor = val;
+}
+
+function syncPointGetReferenceConstructor(): ReferenceConstructor {
+  assert(referenceConstructor, 'Reference.ts has not been loaded');
+  return referenceConstructor;
+}
+
+export class SyncPoint {
```

Comment:
Does this still need to be a class?

Reply:
We can make it an interface that just contains the one member, but that means that some callsites need to be aware of the internals, which used to be private. That's why I left it as is.